# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/bytestreams/AnnexB.h
# Commit: fd44d547e093
# Full Hash: fd44d547e0931ddcb25e9dde2691ad9e99ec2e05
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-26 09:13:45
# Description:
#   Bug 1860840 - return error for invalid hvcc config when converting sample. r=media-playback-reviewers,padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D191794
# ==============================================================================

diff -r 691148c77012 -r fd44d547e093 dom/media/platforms/agnostic/bytestreams/AnnexB.h
--- a/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Wed Oct 25 21:14:33 2023 +0300
+++ b/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Wed Oct 25 17:10:46 2023 +0000
@@ -44,7 +44,8 @@
       mozilla::MediaRawData* aSample,
       const RefPtr<mozilla::MediaByteBuffer>& aAVCCHeader = nullptr);
   // Convert a sample from Annex B to HVCC. An HVCC extradata must not be set.
-  static bool ConvertSampleToHVCC(mozilla::MediaRawData* aSample);
+  static Result<mozilla::Ok, nsresult> ConvertSampleToHVCC(
+      mozilla::MediaRawData* aSample);
 
   // Covert sample to 4 bytes NALU byte stream.
   static mozilla::Result<mozilla::Ok, nsresult> ConvertAVCCTo4BytesAVCC(